From dab49ef97fdf490814e089a01cc9d4bff35f27fc Mon Sep 17 00:00:00 2001
From: Zane Li <zane.li.wiwynn@gmail.com>
Date: Thu, 24 Jul 2025 10:06:07 +0800
Subject: [PATCH] Subject: [PATCH 29/30] i3c: aspeed: Separates the event of 
 ibi and data  consume.

The interrupt resp_q_ready indicate the event of ibi transfer done, data
read by master and master write data. The differnet of these events list
below:
1. if it include the signal ibi_update: ibi transfer done
2. if the response data contain the data length > 0: master write data.
3. others: data read by master.
This patch separates the wait event of the ibi and data consume, making
ISR can signal the appropriate event.

Signed-off-by: Billy Tsai <billy_tsai@aspeedtech.com>
Change-Id: I2f741d2031efc27d1611050db924116983f4ba64
---
 drivers/i3c/i3c_aspeed.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/i3c/i3c_aspeed.c b/drivers/i3c/i3c_aspeed.c
index abb2bba73e8..2a61f997fe4 100644
--- a/drivers/i3c/i3c_aspeed.c
+++ b/drivers/i3c/i3c_aspeed.c
@@ -1787,9 +1787,10 @@ static int i3c_aspeed_slave_wait_data_consume(const struct device *dev)
 	struct i3c_aspeed_obj *obj = DEV_DATA(dev);
 	union i3c_intr_s events;
 
+	osEventFlagsClear(obj->data_event, ~osFlagsError);
 	events.value = 0;
 	events.fields.resp_q_ready = 1;
-	osEventFlagsWait(obj->event_id, events.value, osFlagsWaitAny, osWaitForever);
+	osEventFlagsWait(obj->data_event, events.value, osFlagsWaitAny, osWaitForever);
 
 	return 0;
 }
-- 
2.25.1

